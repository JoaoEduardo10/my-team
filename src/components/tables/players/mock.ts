export const mockPlayers = [
  {
    player: {
      age: 18,
      id: 1,
      name: 'eduardo',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 2,
      name: 'felipe',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 20,
      id: 3,
      name: 'caio',
      nationality: 'brasileiro'
    }
  },
  {
    player: {
      age: 30,
      id: 4,
      name: 'renato',
      nationality: 'brasileiro'
    }
  },
];
