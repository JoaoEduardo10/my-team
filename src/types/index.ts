type TCountry = {
  code: string
  flag: string
  name: string
}

export type TSeason = {
  current: string
  end: string
  start: string
  year: number
}

export type TCountrys = {
  response: TCountry[]
}

export type TTeam = {
  response: [
    {
      team: {
        code: string
        country:string
        id: number
        name:string
        logo: string
      }
    }
  ]
}

export type TLeagues = {
  response: [
    {
      country: TCountry
      league: {
        id: number
        logo: string
        name: string
        type: string
      }
      seasons: TSeason[]
    }
  ]
}

export type TPlayers = {
  response: [
    {
      player: {
        age: number
        id: number
        name: string
        nationality: string
      }
    }
  ]
}

export type TSubFixtures = {
  away: number
  home: number
  total: number
}

export type TSubGoals = {
  percentage: string
  total: number
}

export type TSubLineups = {
  formation: string
  played: number
}

export type TTeamStatistic = {
  response: {
      fixtures: {
        draws: TSubFixtures
        loses: TSubFixtures
        played: TSubFixtures
        wins: TSubFixtures
      },
      goals: {
        against: {
          minute: {
            '0-15': TSubGoals
            '16-30': TSubGoals
            '31-45': TSubGoals
            '46-60': TSubGoals
            '61-75': TSubGoals
            '76-90': TSubGoals
            '91-105': TSubGoals
            '106-120': TSubGoals
        }
        }
      },
      lineups: TSubLineups[],
      team: {
        id: number
        logo: string
        name: string
      }
    }

}
