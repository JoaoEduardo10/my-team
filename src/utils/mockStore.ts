import configureStore from 'redux-mock-store';


const mockStore = configureStore([]);

const store = mockStore({
  team: {
    country: 'brasil',
    season: '2022',
    league: '47',
    team: 'SÃ£o paulo'
  }
});

export { store };
