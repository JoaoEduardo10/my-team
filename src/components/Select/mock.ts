/* eslint-disable @typescript-eslint/no-unused-vars */
import { TCountrys, TLeagues, TTeam } from '../../types';

const mockSeason: number[] = [2022, 2010, 2020];

const mockContrys:TCountrys['response'] = [
  {
    code: 'T1',
    flag: 'test',
    name:  'test 1'
  },
  {
    code: 'T2',
    flag: 'test 2',
    name:  'test 2'
  },
  {
    code: 'T3',
    flag: 'test3',
    name:  'test 3'
  },
];


const mockLeague: TLeagues['response'] = [
  {
    country: {
      code: 'T1',
      flag: 'test',
      name: 'Test 1'
    },
    league: {
      id: 1,
      logo: 'tes.png',
      name: 'test ligue',
      type: 'test'
    },
    seasons: [
      {
        current: 'test',
        end: '10/10/2022',
        start: '01/01/2022',
        year: 2022
      }
    ]
  }
];

const mockTeam: TTeam['response'] = [
  {
    team: {
      code: 'br',
      country: 'brazil',
      id: 1,
      logo: 'test.png',
      name: 'bar√ßa'
    }
  }
];

export { mockContrys, mockLeague, mockSeason, mockTeam };
