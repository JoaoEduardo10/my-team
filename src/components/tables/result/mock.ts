import { TTeamStatistic } from '../../../types';

export const mockResult: TTeamStatistic['response'] = {
  fixtures: {
    draws: {
      away: 4,
      home: 4,
      total: 8
    },
    loses:{
      away: 1,
      home: 4,
      total: 5
    },
    played: {
      away: 1,
      home: 10,
      total: 29
    },
    wins:{
      away: 1,
      home: 5,
      total: 6
    }
  },
  goals: {
    against: {
      minute: {
        '0-15': { percentage: '32.5', total: 2 },
        '106-120': { percentage: '32.5', total: 2 },
        '16-30': { percentage: '32.5', total: 2 },
        '31-45': { percentage: '32.5', total: 2 },
        '46-60': { percentage: '32.5', total: 2 },
        '61-75': { percentage: '32.5', total: 2 },
        '76-90': { percentage: '32.5', total: 2 },
        '91-105': { percentage: '32.5', total: 2 },
      }
    }
  },
  lineups: [
    {
      formation: '4-3-3',
      played: 7
    },
    {
      formation: '4-4-3',
      played: 10
    },
    {
      formation: '4-5-2',
      played: 13
    },
  ],
  team: {
    id: 12,
    logo: 'Saopaulo.png',
    name: 'SÃ£o Paulo'
  }
};
