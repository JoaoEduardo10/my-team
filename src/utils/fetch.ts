/* eslint-disable @typescript-eslint/no-explicit-any */

import { TCountrys, TLeagues, TTeam } from '../types';

const url = import.meta.env.VITE_URL_API as string;

const authUser = async (key: string) => {
  const response = await fetch(`${url}/status`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data = await response.json();

  return data;

};

const getCountry = async (key: string) => {
  const response = await fetch(`${url}/countries`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TCountrys = await response.json();

  return data;
};

const getLeague= async (key: string, country: string, season: number) => {
  const response = await fetch(`${url}/leagues?country=${country}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TLeagues = await response.json();

  return data;
};

const getSeason = async (key: string) => {
  const response = await fetch(`${url}/leagues/seasons`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: { response: number[] } = await response.json();

  return data;
};

const getTeam = async (key: string, season: number, league: number) => {
  const response = await fetch(`${url}/teams?league=${league}&season=${season}`, {
    method: 'GET',
    headers: {
      'x-rapidapi-key': key,
      'X-RapidAPI-Host': 'v3.football.api-sports.io'
    },
    redirect: 'follow'
  });
  const data: TTeam = await response.json();

  return data;
};


export { authUser, getCountry, getLeague, getSeason, getTeam };
