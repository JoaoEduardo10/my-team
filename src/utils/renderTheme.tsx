import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { GlobalStyles } from '../styles/globals-styles';
import { RouterProvider, createBrowserRouter } from 'react-router-dom';
import { AuthProvider } from '../context/auth';
import { Provider } from 'react-redux';
import { store } from './mockStore';

const mockRouet = (children: React.ReactNode, path?: string) => {
  return createBrowserRouter([
    {
      path: path ?? '/',
      element: children
    }
  ]);
};

export const renderTheme = (
  children: React.ReactNode,
  path?: string
) => {
  return render(
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <AuthProvider>
          <RouterProvider router={mockRouet(children, path)} />
        </AuthProvider>
        <GlobalStyles />
      </ThemeProvider>
    </Provider>
  );
};
