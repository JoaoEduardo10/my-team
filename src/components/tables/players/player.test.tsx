import { screen } from '@testing-library/react';
import { Players } from '.';
import { TPlayers } from '../../../types';
import { renderTheme } from '../../../utils/renderTheme';
import { mockPlayers } from './mock';

const newMock: any = [];

describe('<Players />', () => {
  it('should a render Playres table compónent', () => {
    renderTheme(<Players players={mockPlayers as TPlayers['response']} />);

    const heading = screen.getByRole('heading', { name: 'Jogadores' });
    const table = screen.getByRole('table');
    const tHead = screen.getByLabelText('Cabeçalho');
    const data = screen.getAllByLabelText('Dados dos jogadores');
    const alert = screen.queryByLabelText('alert');

    expect(heading).toBeInTheDocument();
    expect(table).toBeInTheDocument();
    expect(tHead).toBeInTheDocument();
    expect(data.length).toEqual(4);
    expect(alert).not.toBeInTheDocument();
  });

  it('should a render Playres table compónent', () => {
    renderTheme(<Players players={newMock} />);

    const alert = screen.getByLabelText('alert');

    expect(alert).toBeInTheDocument();
  });

  it('should match snapshort', () => {
    const { container } = renderTheme(<Players players={mockPlayers as TPlayers['response']} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
