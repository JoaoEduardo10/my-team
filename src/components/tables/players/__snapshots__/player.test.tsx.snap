// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Players /> > should match snapshort 1`] = `
<div
  class="ewBUVk"
>
  <h3
    class="gIwlCh"
  >
    Jogadores
  </h3>
  <table
    class="iHGiYk"
    role="table"
  >
    <thead
      aria-label="CabeÃ§alho"
      class="dDYtZq"
    >
      <tr
        class="hzbqHX"
      >
        <th
          class="czTZEm"
        >
          Nome
        </th>
        <th
          class="czTZEm"
        >
          Idade
        </th>
        <th
          class="czTZEm"
        >
          Nacionalidade
        </th>
      </tr>
    </thead>
    <tbody
      class="fGRXlB"
    >
      <tr
        aria-label="Dados dos jogadores"
        class="hzbqHX"
      >
        <td
          class="cicZYJ"
        >
          eduardo
        </td>
        <td
          class="cicZYJ"
        >
          18
        </td>
        <td
          class="cicZYJ"
        >
          brasileiro
        </td>
      </tr>
      <tr
        aria-label="Dados dos jogadores"
        class="hzbqHX"
      >
        <td
          class="cicZYJ"
        >
          felipe
        </td>
        <td
          class="cicZYJ"
        >
          20
        </td>
        <td
          class="cicZYJ"
        >
          brasileiro
        </td>
      </tr>
      <tr
        aria-label="Dados dos jogadores"
        class="hzbqHX"
      >
        <td
          class="cicZYJ"
        >
          caio
        </td>
        <td
          class="cicZYJ"
        >
          20
        </td>
        <td
          class="cicZYJ"
        >
          brasileiro
        </td>
      </tr>
      <tr
        aria-label="Dados dos jogadores"
        class="hzbqHX"
      >
        <td
          class="cicZYJ"
        >
          renato
        </td>
        <td
          class="cicZYJ"
        >
          30
        </td>
        <td
          class="cicZYJ"
        >
          brasileiro
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
